Source/CesiumRuntime/Private/CesiumGeoreference.cpp

```diff
@@ -378,181 +378,225 @@ void ACesiumGeoreference::PostEditChangeProperty(FPropertyChangedEvent& event) {
-        static_cast<double>(cameraLocation.Z) +
-            static_cast<double>(originLocation.Z),
-        1.0);
-
-    glm::dvec3 cameraECEF = this->_ueAbsToEcef * cameraAbsolute;
-
-    this->_insideSublevel = false;
-
-    const TArray<ULevelStreaming*>& streamedLevels =
-        this->GetWorld()->GetStreamingLevels();
-    for (ULevelStreaming* streamedLevel : streamedLevels) {
-      FString levelName =
-          FPackageName::GetShortName(streamedLevel->GetWorldAssetPackageName());
-      levelName.RemoveFromStart(this->GetWorld()->StreamingLevelsPrefix);
-      // TODO: maybe we should precalculate the level ECEF from level
-      // long/lat/height
-      // TODO: consider the case where we're intersecting multiple level radii
-      for (FCesiumSubLevel& level : this->CesiumSubLevels) {
-        // if this is a known level, we need to tell it whether or not it should
-        // be loaded
-        if (levelName.Equals(level.LevelName)) {
-          glm::dvec3 levelECEF =
-              CesiumGeospatial::Ellipsoid::WGS84.cartographicToCartesian(
-                  CesiumGeospatial::Cartographic::fromDegrees(
-                      level.LevelLongitude,
-                      level.LevelLatitude,
-                      level.LevelHeight));
-
-          if (glm::length(levelECEF - cameraECEF) < level.LoadRadius) {
-            this->_insideSublevel = true;
-            if (!level.CurrentlyLoaded) {
-              this->_jumpToLevel(level);
-              streamedLevel->SetShouldBeLoaded(true);
-              streamedLevel->SetShouldBeVisible(true);
-              level.CurrentlyLoaded = true;
-
-              // If we are not going to continue origin rebasing inside the
-              // sublevel, just set the origin back to zero since the sublevel
-              // will be centered around zero anyways.
-              if (this->KeepWorldOriginNearCamera &&
-                  !this->OriginRebaseInsideSublevels) {
-                this->GetWorld()->SetNewWorldOrigin(FIntVector(0, 0, 0));
-              }
-            }
-          } else {
-            if (level.CurrentlyLoaded) {
-              streamedLevel->SetShouldBeLoaded(false);
-              streamedLevel->SetShouldBeVisible(false);
-              level.CurrentlyLoaded = false;
-            }
-          }
+  return isInsideSublevel;
+}
 
-          break;
-        }
-      }
-    }
+void ACesiumGeoreference::_performOriginRebasing() {
+  bool isGame = this->GetWorld()->IsGameWorld();
+  if (!isGame) {
+    return;
+  }
+  if (!IsValid(WorldOriginCamera)) {
+    return;
+  }
+  if (!this->KeepWorldOriginNearCamera) {
+    return;
+  }
+  if (this->_insideSublevel && !this->OriginRebaseInsideSublevels) {
+    // If we are not going to continue origin rebasing inside the
+    // sublevel, just set the origin back to zero since the sublevel
+    // will be centered around zero anyways.
+    this->GetWorld()->SetNewWorldOrigin(FIntVector(0, 0, 0));
+    return;
+  }
```

Comment:
In this case it will keep redundantly setting the world origin to (0, 0, 0) every Tick. I don't think that's necessarily a problem but it will trigger a bunch of delegates needlessly when it's not supposed to be origin rebasing. I know this is probably the same as the logic beforehand, but we might as well fix it up. I think we should only set origin to zero here if it is not already zero, so maybe we can add a check for that.